src/Sarif.UnitTests/Core/FileDataTests.cs

```diff
@@ -57,24 +57,55 @@ public void FileData_ComputeHashes()
             }
         }
 
-        [Fact]
-        public void FileData_PersistFileContentsBinary()
+        [Theory]
+        // Unknown files are regarded as binary
+        [InlineData(".unknown", LoggingOptions.PersistBinaryContents, true)]
+        [InlineData(".unknown", LoggingOptions.PersistTextFileContents, false)]
+        [InlineData(".exe", LoggingOptions.PersistBinaryContents | LoggingOptions.PersistTextFileContents, true)]
+        [InlineData(".cs",  LoggingOptions.PersistBinaryContents | LoggingOptions.PersistTextFileContents, true)]
```

Comment:
I suggest that your inline data should be
`fileExtension, loggingOptions, textShouldBePersisted, binaryShouldBePersisted`